dart/lib/src/transport/client_report_transport.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:meta/meta.dart';
+import '../../sentry.dart';
+import '../client_reports/client_report_recorder.dart';
+import '../sentry_envelope_header.dart';
+import 'rate_limiter.dart';
+
+/// Decorator that handles attaching of client reports in tandem with rate
+/// limiting. The rate limiter is optional.
```

Comment:
wdym by it's optional? 

I think it's always added to our transports so it's also applying it to native, no?

Reply:
We don't rate limit on the flutter side when using the file based transport. The native libs have this responsibility. In order to reuse for both transports we support now, the injected rate limiter is optional.

Reply:
ah got it

```dart
    if (options.transport is NoOpTransport) {
      rateLimiter = RateLimiter(options);
      options.transport = HttpTransport(options, rateLimiter);
    }
    options.transport = ClientReportTransport(
      rateLimiter,
      options,
      options.transport,
    );
```

was confused at first, to me it looked like at first glance the rateLimiter is always active even for FileSystemTransrpot, didn't realize that it's null there

is it possible to make this a bit more obvious? or maybe add a comment or similar